Task 2/Task 2.1/Task 2.1.2/Menu.cs

```diff
@@ -0,0 +1,278 @@
+                                catch
+                                {
+                                    throw new Exception();
+                                }
+                                break;
+                            case "5":
+                                p1 = InputPoint("Первая");
+                                p2 = InputPoint("Вторая");
+                                p3 = InputPoint("Третья");
+                                p4 = InputPoint("Четвёртая");
+                                try
+                                {
+                                    Rectangle rectangle = new Rectangle(p1, p2, p3, p4);
+                                    user.Figures.Add(rectangle);
+                                    Console.WriteLine($"Создан четырёхугольник:{Environment.NewLine}{rectangle}");
+                                    Console.WriteLine();
+                                    ShowMenu();
+                                }
+                                catch
+                                {
+                                    throw new Exception();
+                                }
+                                break;
+                            case "6":
+                                p1 = InputPoint("Первая");
+                                p2 = InputPoint("Вторая");
+                                p3 = InputPoint("Третья");
+                                p4 = InputPoint("Четвёртая");
+                                try
+                                {
+                                    SquareFigure squareFigure = new SquareFigure(p1, p2, p3, p4);
+                                    user.Figures.Add(squareFigure);
+                                    Console.WriteLine($"Создан квадрат:{Environment.NewLine}{squareFigure}");
+                                    Console.WriteLine();
+                                    ShowMenu();
+                                }
+                                catch
+                                {
+                                    throw new Exception();
+                                }
+                                break;
+                            default:
+                                ShowMenu();
+                                break;
+                        }
+                        break;
+                    }
+                case "2":
+                    int i = 0;
+                    foreach (var item in user.Figures)
+                    {
+                        Console.WriteLine($"Id - {i} {item}");
+                        i++;
+                    }
+                    Console.WriteLine();
+                    ShowMenu();
+                    break;
+                case "3":
+                    Console.Write("Введите id фигуры: ");
+                    int id;
+                    int.TryParse(Console.ReadLine(), out id);
+                    if (user.Figures.Contains(user.Figures.ElementAt(id)))
```

Comment:
```suggestion
                    if (id < user.Figures.Count && id >= 0)
```
Возможно ты имел ввиду вот это, потому что предыдущий вариант падает, при вводе несуществующего id, а вообще, если уж предполагается работа по id, то лучше держать его внутри обьекта